### PR TITLE
`remote modify`: use `--local` where needed

### LINKED ISSUE #2089: cmd: use `--local` in `remote modify` ref
Continuing #2088

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Closes #2089

### CODE REVIEW

File: content/docs/user-guide/setup-google-drive-remote.md [hunk 1]
```diff
@@ -154,7 +154,8 @@ GDrive remote), for example:
 
 ```dvc
 $ dvc remote modify myremote gdrive_client_id 'client-id'
-$ dvc remote modify myremote gdrive_client_secret 'client-secret'
+$ dvc remote modify myremote --local \
```
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:157]: they are per project, not sensitive to that extent

File: content/docs/command-reference/remote/modify.md [hunk 2]
```diff
@@ -557,16 +561,20 @@ more information.
...
 
   ```dvc
-  $ dvc remote modify \
+  $ dvc remote modify --local \
         myremote credentialpath '/home/.../project-XXX.json'
   ```
 
-  Alternatively, the `GOOGLE_APPLICATION_CREDENTIALS` environment variable can
-  be set:
+> The `credentialpath` value may contain sensitive user info. Therefore, it's
```
[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:568]: OK, changed "sensitive" -> "personal".

File: content/docs/user-guide/setup-google-drive-remote.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/setup-google-drive-remote.md]: OK, rolled back this one.

File: content/docs/command-reference/remote/modify.md [hunk 3]
```diff
@@ -464,18 +464,22 @@ a full guide on using Google Drive as DVC remote storage.
   using a custom Google Client project. Also requires using `gdrive_client_id`.
 
   ```dvc
-  $ dvc remote modify myremote \
+  $ dvc remote modify --local myremote \
```
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:467]: here?

File: content/docs/command-reference/remote/modify.md [hunk 4]
```diff
@@ -464,18 +464,22 @@ a full guide on using Google Drive as DVC remote storage.
...
   access Google Drive data. `.dvc/tmp/gdrive-user-credentials.json` by default.
 
   ```dvc
-  $ dvc remote modify myremote gdrive_user_credentials_file \
-                      .dvc/tmp/myremote-credentials.json
+  $ dvc remote modify --local myremote \
+        gdrive_user_credentials_file .dvc/tmp/mycredentials.json
   ```
 
+> Authentication values may contain sensitive user info. Therefore, it's safer
```
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:479]: what values does it apply to now? (jus to check that we have something)

File: content/docs/command-reference/remote/modify.md [hunk 5]
```diff
@@ -557,16 +561,20 @@ more information.
...
-  $ dvc remote modify \
-        myremote credentialpath '/home/.../project-XXX.json'
+  $ dvc remote modify --local myremote \
+          credentialpath '/home/.../project-XXX.json'
   ```
 
-  Alternatively, the `GOOGLE_APPLICATION_CREDENTIALS` environment variable can
-  be set:
+> The `credentialpath` value may contain personal user info. Therefore, it's
+> safer to use the `--local` flag so it's written to a Git-ignored
```
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:569]: safer -> convenient or makes sense

File: content/docs/command-reference/remote/modify.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:467]: Yes? (Not sure what the Q was, sorry.) This is for `gdrive_client_secret`.

File: content/docs/command-reference/remote/modify.md (hunk 4 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:479]: gdrive_client_secret and gdrive_user_credentials_file

File: content/docs/command-reference/remote/modify.md (hunk 3 above)
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:467]: `gdrive_client_secret` is also fine to include

File: content/docs/command-reference/remote/modify.md (hunk 4 above)
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:479]: they are both not sensitive

**Review by shcheklein — APPROVED** [review]

**Pull request merged**